test/functional/JLM_Tests/src/org/openj9/test/java/lang/management/TestOpenJ9DiagnosticsMXBean.java

```diff
@@ -263,10 +264,30 @@ private void testLocal_setDumpOptions() {
 			e.printStackTrace();
 		}		
 
+		dumpFileName = "javacore_unsupported.txt";
+		dumpFilePath = dir + File.separator + dumpFileName;
+		int counter = 1;
+		while(true) {
+			try {
+				Thread.sleep(100);
+				diagBean.setDumpOptions("java:events=catch,filter=java/io/UnsupportedEncodingException,range=1..1,file=" + dumpFilePath);
+				break;
+			} catch (ConfigurationUnavailableException e) {
+				// Ignore exception and try setting options again
+			} catch (InvalidOptionException e) {
+				e.printStackTrace();
+                                Assert.fail("Unexpected exception thrown: " + e.getMessage());
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+				Assert.fail("Exception occurred while sleeping thread: " + e.getMessage());
+			}
+			if (counter == 20) {
+				break;
+			} 
+			counter++;
+		}
```

Comment:
We don't want to do the sleep the first time if we can avoid it.  The sleep is intended to be a fallback only if required.

Something like this:
```java
while(true) {
	try { 
		diagBean.setDumpOptions("java:events=catch,filter=java/io/UnsupportedEncodingException,range=1..1,file=" + dumpFilePath);
		break;
	} catch (ConfigurationUnavailableException e) {
		// Ignore exception and try setting options again
		try {
			Thread.sleep(100);
		} catch (InterruptedException e) {
			// Do nothing - the outer loop will retry if required
		}
		// Only attempt this 20 times before failing the test
		if (counter == 20) {
			e.printStackTrace();
			Assert.fail("Unable to update the dump options: " + e.getMessage());
		} 
		counter++;
	} catch (InvalidOptionException e) {
		e.printStackTrace();
		Assert.fail("Unexpected exception thrown: " + e.getMessage());
	}
}
```

The `} catch (InvalidOptionException e) {` block is better handled by letting the test throw the InvalidOperationException and having TestNG handle it.  Letting the test system deal with unexpected exceptions means you get consistent output from failed tests rather than System.out.printlns

Reply:
@DanHeidinga Thanks, I have made the changes as suggested. I have also included passing of mode to the remote process and re-used ProcessLocking test class to synchronize between both the processes. Please review again.